src/ports/nfts/queries.ts

```diff
@@ -1,10 +1,159 @@
+  if (!rarities || !rarities.length) {
+    return null
+  }
+
+  return SQL` (nft.search_wearable_rarity = ANY (${rarities}) OR nft.search_emote_rarity = ANY (${rarities})) `
+}
+
+function getNFTWhereStatement(nftFilters?: NFTFilters): SQLStatement {
+  if (!nftFilters) {
+    return SQL``
+  }
+
+  const FILTER_BY_CATEGORY = nftFilters.category ? SQL` LOWER(nft.category) = LOWER(${nftFilters.category}) ` : null
+  const FILTER_BY_OWNER = nftFilters.owner ? SQL` LOWER(account.address) = LOWER(${nftFilters.owner}) ` : null
+  const FILTER_BY_TOKEN_ID = nftFilters.tokenId ? SQL` nft.token_id = ${nftFilters.tokenId} ` : null
+  const FILTER_BY_ITEM_ID = nftFilters.itemId ? SQL` LOWER(nft.item_id) = LOWER(${nftFilters.itemId}) ` : null
+  const FILTER_BY_NETWORK = nftFilters.network ? SQL` nft.network = ANY (${getDBNetworks(nftFilters.network)}) ` : null
+  const FILTER_BY_HAS_SOUND = nftFilters.emoteHasSound ? SQL` emote.has_sound = true ` : null
+  const FILTER_BY_HAS_GEOMETRY = nftFilters.emoteHasGeometry ? SQL` emote.has_geometry = true ` : null
+  const FILTER_MIN_ESTATE_SIZE = nftFilters.minEstateSize ? SQL` estate.size >= ${nftFilters.minEstateSize} ` : null
+  const FILTER_MAX_ESTATE_SIZE = nftFilters.maxEstateSize ? SQL` estate.size <= ${nftFilters.maxEstateSize} ` : null
+  const FILTER_BY_WEARABLE_CATEGORY = nftFilters.wearableCategory ? SQL` wearable.category = ${nftFilters.wearableCategory} ` : null
+  const FILTER_BY_EMOTE_CATEGORY = nftFilters.emoteCategory ? SQL` emote.category = ${nftFilters.emoteCategory} ` : null
+  const FILTER_BY_WEARABLE_HEAD = nftFilters.isWearableHead ? SQL` nft.search_is_wearable_head = true ` : null
+  const FILTER_BY_LAND = nftFilters.isLand ? SQL` nft.search_is_land = true ` : null
+  const FILTER_BY_WEARABLE_ACCESSORY = nftFilters.isWearableAccessory ? SQL` nft.search_is_wearable_accessory = true ` : null
+  const FILTER_BY_WEARABLE_SMART = nftFilters.isWearableSmart ? SQL` nft.item_type = ${ItemType.SMART_WEARABLE_V1} ` : null
+  const FILTER_BY_CONTRACT_ADDRESS =
+    nftFilters.contractAddresses && nftFilters.contractAddresses.length
+      ? SQL` nft.contract_address = ANY (${nftFilters.contractAddresses}) `
+      : null
+  const FILTER_BY_TEXT = nftFilters.search ? SQL` nft.search_text % ${nftFilters.search} ` : null
+  const FILTER_BY_MIN_DISTANCE_TO_PLAZA = nftFilters.minDistanceToPlaza
+    ? SQL` nft.search_distance_to_plaza >= ${nftFilters.minDistanceToPlaza} `
+    : null
+  const FILTER_BY_MAX_DISTANCE_TO_PLAZA = nftFilters.maxDistanceToPlaza
+    ? SQL` nft.search_distance_to_plaza <= ${nftFilters.maxDistanceToPlaza} `
+    : null
+  const FILTER_BY_ADJACENT_TO_ROAD = nftFilters.adjacentToRoad ? SQL` nft.search_adjacent_to_road = true ` : null
+  const FILTER_BY_EMOTE_PLAY_MODE = getEmotePlayModeWhereStatement(nftFilters.emotePlayMode)
+  const FILTER_BY_EMOTE_GENDERS = getGenderWhereStatement(true, nftFilters.emoteGenders)
```

Comment:
do you know if the emote gender filter is required? emotes are unisex

Reply:
It is available right now in the nft-server 🤔 not sure why. I would keep it just in case to have the same functionality